keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,235 @@
+            raise ValueError("`threshold` must be sorted, got {}".format(thresholds))
+        self.match_values = match_values
+        if len(match_values) != len(thresholds) + 1:
+            raise ValueError(
+                "len(`match_values`) must be len(`thresholds`) + 1, got "
+                "match_values {}, thresholds {}".format(match_values, thresholds)
+            )
+        thresholds = thresholds[:]
+        thresholds.insert(0, -float("inf"))
+        thresholds.append(float("inf"))
+        self.thresholds = thresholds
+        self._force_match_for_each_col = force_match_for_each_col
+
+    def __call__(self, similarity_matrix: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
+        """Matches each row to a column based on argmax
+
+        Args:
+          similarity_matrix: A float Tensor of shape [num_rows, num_cols] or
+            [batch_size, num_rows, num_cols] representing any similarity metric.
+
+        Returns:
+          matched_columns: An integer tensor of shape [num_rows] or [batch_size,
+            num_rows] storing the index of the matched colum for each row.
+          matched_values: An integer tensor of shape [num_rows] or [batch_size,
+            num_rows] storing the match result (positive match, negative match,
+            ignored match).
+        """
+        squeeze_result = False
+        if len(similarity_matrix.shape) == 2:
+            squeeze_result = True
+            similarity_matrix = tf.expand_dims(similarity_matrix, axis=0)
+        static_shape = similarity_matrix.shape.as_list()
+        num_rows = static_shape[1] or tf.shape(similarity_matrix)[1]
+        batch_size = static_shape[0] or tf.shape(similarity_matrix)[0]
+
+        def _match_when_cols_are_empty():
+            """Performs matching when the rows of similarity matrix are empty.
+            When the rows are empty, all detections are false positives. So we return
+            a tensor of -1's to indicate that the rows do not match to any columns.
+            Returns:
+                matched_columns: An integer tensor of shape [num_rows] or [batch_size,
+                num_rows] storing the index of the matched column for each row.
+                matched_values: An integer tensor of shape [num_rows] or [batch_size,
+                num_rows] storing the match type indicator (e.g. positive or negative
+                or ignored match).
+            """
+            with tf.name_scope("empty_gt_boxes"):
+                matched_columns = tf.zeros([batch_size, num_rows], dtype=tf.int32)
+                matched_values = -tf.ones([batch_size, num_rows], dtype=tf.int32)
+                return matched_columns, matched_values
+
+        def _match_when_cols_are_non_empty():
+            """Performs matching when the rows of similarity matrix are non empty.
+            Returns:
+                matched_columns: An integer tensor of shape [num_rows] or [batch_size,
```

Comment:
Since you expand the batch dim, lets just stay with [batch_size, num_rows] here.

Reply:
Done.